src/detect-mqtt-connect-flags.c

```diff
@@ -134,30 +134,36 @@ static int DetectMQTTConnectFlagsMatch(DetectEngineThreadCtx *det_ctx,
 static DetectMQTTConnectFlagsData *DetectMQTTConnectFlagsParse(const char *rawstr)
 {
     DetectMQTTConnectFlagsData *de = NULL;
+    char *copy = NULL;
     int ret = 0;
     int ov[MAX_SUBSTRINGS];
 
     ret = DetectParsePcreExec(&parse_regex, rawstr, 0, 0, ov, MAX_SUBSTRINGS);
     if (ret < 1) {
         SCLogError(SC_ERR_PCRE_MATCH, "invalid flag definition: %s", rawstr);
-        return NULL;
+        goto error;
```

Comment:
I feel this use of the error label is adding unnecessary complexity. Its not going to do anything there, but now we need to follow this jump logic to realize this.

Reply:
This is just to address Coverity's complaint about us checking `de` for `NULL` in the previous code -- it thinks that it might be NULL but it has been dereferenced before. Apparently it didn't catch that in that case we already returned from the function before in that case.

Reply:
I think it tells you there is an issue: by checking it can be NULL, it assumes the programmer "knows" that it can be. But in all the paths leading to the NULL check its already dereferenced. So it alerts you that you may have a bug before the null check. Of course it were smarter it would see that `de` was already checked for null.

Reply:
Got it. So you agree we should not try to work around that and remove the check in these few cases?

Reply:
See #5261